### PR TITLE
Use frozen::unordered_map for ErrorMessages

### PR DESCRIPTION
**Pull request by goedderz**
### Scope & Purpose

As error codes and messages are known at compile time, it makes sense to save them in a compile-time map, rather than a runtime map.

- [ ] :hankey: Bugfix (requires CHANGELOG entry)
- [ ] :pizza: New feature (requires CHANGELOG entry, feature documentation and release notes)
- [ ] :fire: Performance improvement
- [X] :hammer: Refactoring/simplification
- [ ] :book: CHANGELOG entry made

#### Backports:

- [X] No backports required
- [ ] Backports required for: *(Please specify versions and link PRs)*

### Testing & Verification

- [X] This change is a trivial rework / code cleanup without any test coverage.
- [ ] The behavior in this PR was *manually tested*
- [ ] This change is already covered by existing tests, such as *(please describe tests)*.
- [ ] This PR adds tests that were used to verify all changes:
  - [ ] Added new C++ **Unit tests**
  - [ ] Added new **integration tests** (e.g. in shell_server / shell_server_aql)
  - [ ] Added new **resilience tests** (only if the feature is impacted by failovers)
- [ ] There are tests in an external testing repository:
- [ ] I ensured this code runs with ASan / TSan or other static verification tools

http://jenkins.arangodb.biz:8080/job/arangodb-matrix-pr/13868/

### CODE REVIEW

**Comment by goedderz** [comment]
image

File: utils/generateErrorfile.py [hunk 1]
```diff
@@ -107,19 +111,39 @@ def genCHeaderFile(errors):
            + "/// " + e[1] + ": " + e[0] + "\n"\
            + wrap(e[2], 80, 0, 0, "/// \"") + "\"\n"\
            + wrap(e[3], 80, 0, 0, "/// ") + "\n"\
-           + "constexpr auto TRI_" + e[0].ljust(61) + " = ErrorCode{" + e[1] + "};\n"\
-           + "\n"
-
-  header = header + "\n"\
-           + "/// register all errors for ArangoDB\n"\
-           + "void TRI_InitializeErrorMessages();\n"\
+           + "constexpr auto " + errorName(e).ljust(61) + " = ErrorCode{" + e[1] + "};\n"\
```
Comment (goedderz) [inline utils/generateErrorfile.py:114]:
```suggestion
           + "constexpr auto " + errorName(e).ljust(65) + " = ErrorCode{" + e[1] + "};\n"\
```

**Comment by goedderz** [comment]
http://jenkins.arangodb.biz:8080/job/arangodb-matrix-pr/13896/

**Comment by goedderz** [comment]
image

**Comment by goedderz** [comment]
http://jenkins.arangodb.biz:8080/job/arangodb-matrix-pr/13927/

**Comment by goedderz** [comment]
image

**Review by jsteemann — APPROVED** [review]

**Pull request merged**